Lib/pdb.py

```diff
@@ -414,16 +416,35 @@ def preloop(self):
                     self.message('display %s: %r  [old: %r]' %
                                  (expr, newvalue, oldvalue))
 
-    def interaction(self, frame, traceback):
+    def interaction(self, frame, tb_or_exc):
         # Restore the previous signal handler at the Pdb prompt.
+
+        if isinstance(tb_or_exc, BaseException):
+            tb_or_exc, exception = tb_or_exc.__traceback__, tb_or_exc
+            self._chained_exceptions = [exception]
+            self._chained_exception_index = 0
+            open_list = [exception]
+            while open_list:
+                current = open_list.pop(0)
+                for ex_type in ["__cause__", "__context__"]:
```

Comment:
In the standard traceback you get the context only if there is no cause. I think it makes sense here also to include the context only when there is no cause. See https://github.com/python/cpython/blob/fb08b7905eaf67902a89eeec0f8c1aa5ee0c7ea5/Lib/traceback.py#L740 .

Reply:
Sure, changed to get into `__cause__` if present, otherwise `__context__` assuming `__supress_context__ == False`. Thus this should now work with `raise from None` (test added).

I tend to think that when using `pm()` you sort of really want to get to the root cause  of things, and it might be useful to get not have access to hidden context or cause. But I guess that can be added later if we wish.